src/metadataTextTracks.js

```diff
@@ -0,0 +1,123 @@
+import videojs from 'video.js';
+
+const metadataTextTracks = {};
+let includedCues = {};
+
+/**
+* This feature allows metadata text tracks to be manipulated once they are available,
+* usually after the 'loadstart' event is observed on the player
+* @param player A reference to a player
+* @param processTrack A method that performs some operations on a metadata text track
+**/
+metadataTextTracks.process = function(player, processTrack) {
+  const tracks = player.textTracks();
+  const prepareTrack = function(track) {
+    if (track.kind === 'metadata') {
+      // Make sure track is 'hidden' rather than 'disabled' as it may have already
+      // existed and had been 'disabled' during a playlist change
+      track.mode = 'hidden';
+      processTrack(player, track);
+    }
+  };
+
+  // Text tracks are available
+  if (tracks.length > 0) {
+    for (let i = 0; i < tracks.length; i++) {
+      const track = tracks[i];
+      prepareTrack(track);
+    }
+  // Wait until text tracks are added
+  } else {
+    tracks.addEventListener('addtrack', (event) => {
+      const track = event.track;
+      prepareTrack(track);
+    });
+  }
+};
+
+/**
+* @override only if necessary
```

Comment:
I'm not sure the override messages are useful, my inclination would be to leave them out.